src/main/java/net/minecraftforge/fluids/FluidTank.java

```diff
@@ -175,11 +198,38 @@ public FluidStack drain(int maxDrain, boolean doDrain)
                 fluid = null;
             }
 
+            onContentsChanged();
+
             if (tile != null)
             {
                 FluidEvent.fireEvent(new FluidEvent.FluidDrainingEvent(fluid, tile.getWorld(), tile.getPos(), this, drained));
             }
         }
         return stack;
     }
+
+    /**
+     * Returns true if the tank can be filled with this type of fluid.
+     * Used as a filter for fluid types.
+     * Does not consider the current state of the tank, only whether it could ever fill with this type of fluid.
+     */
+    public boolean canFillFluidType(FluidStack fluid)
+    {
```

Comment:
Why are these no longer part of IFluidHandler? Do you plan to replace them with the improved TankInfo stuff?